lib/hotel_block.rb

```diff
@@ -0,0 +1,96 @@
+require 'date'
+# # require_relative 'hotel.rb'
+
+module Booking
+  class Block
+
+    attr_reader :checkin_date, :checkout_date, :requested_rooms, :discounted_rate, :party_name, :room_block, :block_of_rooms, :all_rooms, :max_rooms, :room_numbers, :hotel, :all_reservations
+    
+    def initialize(checkin_date, checkout_date, requested_rooms, discounted_rate, hotel: nil)
+    @checkin_date = checkin_date
+    @checkout_date = checkout_date 
+    @block_of_rooms ||= []
+    @hotel ||= hotel
```

Comment:
Similar to above, this is saying that if `@hotel` is `nil` (which it is), then assign it to the value of the local variable `hotel` (which has a default value of `nil` if nothing is passed in!), so it's probably okay to just say `@hotel = hotel`